src/UberStrok.Realtime.Server.Game/DeathMatchGameRoom.cs

```diff
@@ -0,0 +1,83 @@
+﻿using log4net;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using UberStrok.Core.Common;
+using UberStrok.Core.Views;
+
+namespace UberStrok.Realtime.Server.Game
+{
+    class DeathMatchGameRoom : BaseGameRoom
+    {
+        private readonly static ILog s_log = LogManager.GetLogger(nameof(DeathMatchGameRoom));
+        private GameRoomDataView gameRoomDataView;
+        public Dictionary<int, int> PlayerScores;
+
+        public DeathMatchGameRoom(GameRoomDataView data) : base(data)
+        {
+            if (data.GameMode != GameModeType.DeathMatch)
+                throw new ArgumentException("GameRoomDataView is not in deathmatch mode.", nameof(data));
+            gameRoomDataView = data;
+            PlayerScores = new Dictionary<int, int>();
+        }
+
+        protected override void OnPlayerJoined(PlayerJoinedEventArgs args)
+        {
+            base.OnPlayerJoined(args);
+
+            if (PlayerScores.ContainsKey(args.Player.Actor.Cmid))
+            {
+                PlayerScores.Remove(args.Player.Actor.Cmid);
+            }
+
+            PlayerScores.Add(args.Player.Actor.Cmid, 0);
+
+            //TODO: Send remaining kills
+            args.Player.Events.Game.SendKillsRemaining(GetKillsRemaining(), GetLeaderCmid());
+        }
+
+        protected override void OnPlayerLeft(PlayerLeftEventArgs args)
+        {
+            base.OnPlayerLeft(args);
+
+            if (PlayerScores.ContainsKey(args.Player.Actor.Cmid))
+            {
+                PlayerScores.Remove(args.Player.Actor.Cmid);
+            }
+
+            //TODO: Send remaining kills
+            int killsRemaining = GetKillsRemaining();
+            int leaderCmid = GetLeaderCmid();
+
+            foreach (var player in Players)
+                player.Events.Game.SendKillsRemaining(killsRemaining, leaderCmid);
+        }
+
+        protected override void OnPlayerKilled(PlayerKilledEventArgs args)
+        {
+            base.OnPlayerKilled(args);
+
+            /* If player killed himself, don't update round score. */
+            if (args.AttackerCmid == args.VictimCmid)
+                return;
+
+            PlayerScores[args.AttackerCmid]++;
+
+            int killsRemaining = GetKillsRemaining();
+            int leaderCmid = GetLeaderCmid();
+
+            foreach (var player in Players)
+                player.Events.Game.SendKillsRemaining(killsRemaining, leaderCmid);
+        }
+
+        private int GetKillsRemaining()
+        {
+            return gameRoomDataView.KillLimit - PlayerScores.Values.Max();
+        }
+
+        private int GetLeaderCmid()
+        {
+            return PlayerScores.Aggregate((a, b) => a.Value > b.Value ? a : b).Key;
```

Comment:
The game doesn't seem to care about this value, so I think we should simply ignore it and not even bother finding the proper value.